lambda-hot-reloading/javascript-terraform-layers/README.md

```diff
@@ -0,0 +1,115 @@
+# LocalStack Demo: Hot code swapping for Lambda functions using LocalStack’s code mounting in JavaScript
+
+## Prerequisites
+
+* LocalStack Pro
+* Docker
+* [awslocal](https://github.com/localstack/awscli-local) CLI
+* Terraform
+* [tflocal](https://github.com/localstack/terraform-local) CLI
+
+## Introduction to the sample
+In this sample, we demonstrate a hot reloading setup where both the function code, and the layer code, is hot reloaded.
+Any changes to either of those two directories will reload the function, and the changed code will be available almost immediately.
+
+We will again use the sample from our [javascript](../javascript/), but one of the values we want to change is supplied by a function defined in our layer.
+
+We will use terraform to deploy a hot reloaded lambda function, and then interact with it by invoking it and changing its source.
+
+The source code of the created lambda function `hotreloadlambda` will is located in the subfolder [lambda_src](./lambda_src/) and of the created layer `hot_reload_layer` it is located in the subfolder [layer_src](./layer_src/).
+
+
+## Starting up
+
+First, we need to make sure we start LocalStack with the right configuration. 
+Hot reloading of layers is only supported in our new lambda provider, all you need to do is set `PROVIDER_OVERRIDE_LAMBDA=v2`, if you use a LocalStack version < 2.0.
+
+```bash
+PROVIDER_OVERRIDE_LAMBDA=v2 localstack start
+```
+
+Accordingly, if you are launching LocalStack via Docker or Docker Compose:
+
+```bash
+#docker-compose.yml
+
+services:
+  localstack:
+    ...
+    environment:
+      ...
+      - PROVIDER_OVERRIDE_LAMBDA=v2
+```
+
+## Deploying
+
+Now we can deploy our terraform stack by using `tflocal`.
+First, we initialize the terraform working directory using:
+
+```bash
+tflocal init
+```
+
+Afterwards, we can deploy our stack on LocalStack:
+
+```bash
+tflocal apply
+```
+
+The terraform configuration will automatically deploy the lambda with hot reloading for the function code.
+The function code will be the contents of the `lambda_src` subdirectory and the layer code in the `layer_src` subdirectory.
+
+## Invoking the Lambda function
+
+We can quickly make sure that it works by invoking it with a simple payload:
+
+```bash
+awslocal lambda invoke --function-name hotreloadlambda output.txt
+```
```

Comment:
might need a v1 vs. v2 aws-cli disclaimer somewhere

Reply:
Should not be needed here as we do not send a payload anyway.